File: src/scene.rs
use camera::Camera;
use background::Background;
use hitable::Hitable;

use std::vec::Vec;

pub struct Scene {
    pub camera: Camera,
    pub background: Box<Background + Send + Sync>,
    pub object_list: Vec<Box<Hitable + Send + Sync>>
}

impl Scene {
    pub fn new(camera: &Camera,
               background: Box<Background + Send + Sync>,
               object_list: Vec<Box<Hitable + Send + Sync>>) -> Scene {
        Scene {
            camera: *camera,
            background: background,
            object_list: object_list
        }
    }
}

File: scene/scene.py
from background import *
from camera import *
from emitter import *
from metal import *
from rectangle import *
from sphere import *
from lambertian import *
from phong import *
from triangle_mesh import *
from materials import *

import json

class Scene:
    def __init__(self):
        self.camera = None
        self.object_list = []
        self.background = None
        
    def add_object(self, obj):
        self.object_list.append(obj)

    def write(self):
        print(json.dumps({
            "background": self.background,
            "camera": self.camera,
            "object_list": self.object_list
            }))
            
